diff -uN version-17a_17.1/include/foruser/NYCgeo.h version-17a_17.1/include/fixed/NYCgeo.h
--- version-17a_17.1/include/foruser/NYCgeo.h	2017-02-06 12:32:16.000000000 -0500
+++ version-17a_17.1/include/fixed/NYCgeo.h	2017-03-12 13:47:59.000000000 -0400
@@ -1,16 +1,19 @@
-//---------------------------------------------------------------------------
 #ifndef NYCgeoH
 #define NYCgeoH
 
 #ifdef __cplusplus
-extern "C" 
+extern "C"
 {
-	__declspec(dllimport) void __stdcall NYCgeo(char *ptr_wa1, char *ptr_wa2=NULL);
-}
+#endif
+
+#ifdef __cplusplus
+	__declspec(dllimport) void __stdcall NYCgeo(char *ptr_wa1, char *ptr_wa2=null);
 #else
-{
 	__declspec(dllimport) void __stdcall NYCgeo(char *ptr_wa1, char *ptr_wa2);
+#endif
+
+#ifdef __cplusplus
 }
 #endif
+
 #endif
-//---------------------------------------------------------------------------
diff -uN version-17a_17.1/include/foruser/geo.h version-17a_17.1/include/fixed/geo.h
--- version-17a_17.1/include/foruser/geo.h	2017-02-06 12:32:16.000000000 -0500
+++ version-17a_17.1/include/fixed/geo.h	2017-03-12 13:56:36.000000000 -0400
@@ -1,25 +1,25 @@
-//---------------------------------------------------------------------------
-
 #ifndef NYCgeoH
 #define NYCgeoH
-//---------------------------------------------------------------------------
 
-////////////////////////////////////////////////////////////////////////////////
-//*** below is the declaration for __stdcall (pascal) 
-////////////////////////////////////////////////////////////////////////////////
-#ifdef WIN32 
-#ifdef __cplusplus                                                      
-extern "C" {                                                            
-#endif
-//void __declspec(dllimport) __stdcall geo(char *ptr_wa1, char *ptr_wa2);
-//void ROLE __stdcall geo(char *ptr_wa1, char *ptr_wa2);
-void ROLE geo(char *ptr_wa1, char *ptr_wa2);
+/*
+ * 2016-01-04 - Matthew Lipper
+ *
+ * Re-wrote this header to fix broken macro and Linux function definition.
+ *
+ */
+
 #ifdef __cplusplus
-       }
+extern "C" {
+#endif
+
+#ifdef _WIN32
+  void __declspec(dllimport) __stdcall geo(char *ptr_wa1, char *ptr_wa2);
+#else
+  void geo(char *ptr_wa1, char *ptr_wa2);
 #endif
-//---------------------------------------------------------------------------
+
+#ifdef __cplusplus
+}
 #endif
-#elif defined (__linux__) 
-extern void geo(char *ptr_wa1, char *ptr_wa2);
-#endif  
- 
+
+#endif // NYCgeoH
diff -uN version-17a_17.1/include/foruser/pac.h version-17a_17.1/include/fixed/pac.h
--- version-17a_17.1/include/foruser/pac.h	2017-02-06 12:32:16.000000000 -0500
+++ version-17a_17.1/include/fixed/pac.h	2017-03-12 14:16:52.000000000 -0400
@@ -27,9 +27,9 @@
 #define UNITT_SIZE     4               //unit type            v16.4
 #define UNITI_SIZE    10               //unit identfier       v16.4
 
-typedef struct { char boro;              // Borough Code              
-                 char SC10[10];          // 10 Digit Street Code     
-                 char Street_name[32];   // Street Name      
+typedef struct { char boro;              // Borough Code
+                 char SC10[10];          // 10 Digit Street Code
+                 char Street_name[32];   // Street Name
                } STREET;
 
 typedef union { char bbl[10];            /* Borough-Block-Lot        */
@@ -104,7 +104,7 @@
                  char attrbytes[3];      /* NAP Identification Number*/
                  char reason_code_2;     /* 2nd Reason Code          */
                  char reason_code_qual_2;/* 2nd Reason Code Qualifier*/
-             //  char filler08_2;         * Future Use            
+             //  char filler08_2;         * Future Use
                  char warn_code_2[2];    /* 2nd Warning Return Code  */
                  char ret_code_2[2];     /* 2nd GeoSupport Return Cod*/
                  char msg_2[80];         /* 2nd GeoSupport Message   */
@@ -116,7 +116,7 @@
                  char int_use1;          /* Internal Use Only        */
                  char reason_code;       /* Reason Code              */
                  char reason_code_qual;  /* Reason Code Qualifier    */
-             //  char filler08;           ** Future Use            
+             //  char filler08;           ** Future Use
                  char warn_code[2];      /* Warning Ret. Code-NotImpl*/
                  char ret_code[2];       /* GeoSupport Return Code   */
                  char msg[80];           /* GeoSupport Message       */
@@ -359,7 +359,7 @@
                                          /* Priority (P,S,T,V)       */
                  char sanit_org_pick_up[5];/* Organics Pick up       */
                  char sanit_bulk_pick_up[5]; /* Bulk Pick Up V16.4   */
-               //char sanit_reserved[5]; 
+               //char sanit_reserved[5];
                  char hurricane_zone[2]; /* Hurricane Evacuation Zone*/
                  char filler04[11];      /* Future Use               */
                  char true_hns[11];      /* Underlying HNS           */
diff -uN version-17a_17.1/include/foruser/wa_fields.h version-17a_17.1/include/fixed/wa_fields.h
--- version-17a_17.1/include/foruser/wa_fields.h	2017-02-06 12:32:16.000000000 -0500
+++ version-17a_17.1/include/fixed/wa_fields.h	2017-03-12 14:18:46.000000000 -0400
@@ -1,89 +1,89 @@
-#if !defined(WA2FIELDS) 
-#define WA2FIELDS 
+#if !defined(WA2FIELDS)
+#define WA2FIELDS
 
-typedef struct _FIELDENTRY { 
-  char fld_name[32]; 
-  int  fld_len; 
-  int  fld_start; 
-  int  arr_entries; 
-  int  arr_len; 
-  int  start_structure; 
-  int  end_structure; 
-  char data_type; 
-  char entry_type; 
+typedef struct _FIELDENTRY {
+  char fld_name[32];
+  int  fld_len;
+  int  fld_start;
+  int  arr_entries;
+  int  arr_len;
+  int  start_structure;
+  int  end_structure;
+  char data_type;
+  char entry_type;
 } FIELDENTRY, *PFIELDENTRY;
 
 //spo(2012-12-17):  I don't think I need extern if not using dll.
-extern PFIELDENTRY wa1_flds; 
+extern PFIELDENTRY wa1_flds;
 extern int n_wa1_field_recs;
 
-extern PFIELDENTRY wa2_f1_flds; 
+extern PFIELDENTRY wa2_f1_flds;
 extern int n_wa2_f1_field_recs;
 
-extern PFIELDENTRY wa2_f1a_flds; 
+extern PFIELDENTRY wa2_f1a_flds;
 extern int n_wa2_f1a_field_recs;
 
-extern PFIELDENTRY wa2_f1ax_flds; 
+extern PFIELDENTRY wa2_f1ax_flds;
 extern int n_wa2_f1ax_field_recs;
 
-extern PFIELDENTRY wa2_f1b_flds; 
+extern PFIELDENTRY wa2_f1b_flds;
 extern int n_wa2_f1b_field_recs;
 
-extern PFIELDENTRY wa2_f1e_flds; 
+extern PFIELDENTRY wa2_f1e_flds;
 extern int n_wa2_f1e_field_recs;
 
-extern PFIELDENTRY wa2_f1ex_flds; 
+extern PFIELDENTRY wa2_f1ex_flds;
 extern int n_wa2_f1ex_field_recs;
 
-extern PFIELDENTRY wa2_f2_flds; 
+extern PFIELDENTRY wa2_f2_flds;
 extern int n_wa2_f2_field_recs;
 
-extern PFIELDENTRY wa2_f2w_flds; 
+extern PFIELDENTRY wa2_f2w_flds;
 extern int n_wa2_f2w_field_recs;
 
-extern PFIELDENTRY wa2_f3_flds; 
+extern PFIELDENTRY wa2_f3_flds;
 extern int n_wa2_f3_field_recs;
 
-extern PFIELDENTRY wa2_f3as_flds; 
+extern PFIELDENTRY wa2_f3as_flds;
 extern int n_wa2_f3as_field_recs;
 
-extern PFIELDENTRY wa2_f3c_flds; 
+extern PFIELDENTRY wa2_f3c_flds;
 extern int n_wa2_f3c_field_recs;
 
-extern PFIELDENTRY wa2_f3cas_flds; 
+extern PFIELDENTRY wa2_f3cas_flds;
 extern int n_wa2_f3cas_field_recs;
 
-extern PFIELDENTRY wa2_f3cx_flds; 
+extern PFIELDENTRY wa2_f3cx_flds;
 extern int n_wa2_f3cx_field_recs;
 
-//extern PFIELDENTRY wa2_f3cxas_flds; 
+//extern PFIELDENTRY wa2_f3cxas_flds;
 //extern int n_wa2_f3xas_field_recs;
 
-extern PFIELDENTRY wa2_f3x_flds; 
+extern PFIELDENTRY wa2_f3x_flds;
 extern int n_wa2_f3x_field_recs;
 
-extern PFIELDENTRY wa2_f3s_flds; 
+extern PFIELDENTRY wa2_f3s_flds;
 extern int n_wa2_f3s_field_recs;
 
-extern PFIELDENTRY wa2_fbl_flds; 
+extern PFIELDENTRY wa2_fbl_flds;
 extern int n_wa2_fbl_field_recs;
 
-extern PFIELDENTRY wa2_fblx_flds; 
+extern PFIELDENTRY wa2_fblx_flds;
 extern int n_wa2_fblx_field_recs;
 
-extern PFIELDENTRY wa2_fbn_flds; 
+extern PFIELDENTRY wa2_fbn_flds;
 extern int n_wa2_fbn_field_recs;
 
-extern PFIELDENTRY wa2_fbnx_flds; 
+extern PFIELDENTRY wa2_fbnx_flds;
 extern int n_wa2_fbnx_field_recs;
 
-extern PFIELDENTRY wa2_fhr_flds; 
+extern PFIELDENTRY wa2_fhr_flds;
 extern int n_wa2_fhr_field_recs;
 
-extern PFIELDENTRY wa2_fap_flds; 
+extern PFIELDENTRY wa2_fap_flds;
 extern int n_wa2_fap_field_recs;
 
-extern PFIELDENTRY wa2_fapx_flds; 
+extern PFIELDENTRY wa2_fapx_flds;
 extern int n_wa2_fapx_field_recs;
 
-#endif 
+#endif
